File: fbi_engine_analyse/src/com/fr/bi/cal/analyze/report/report/widget/chart/newstyle/filter/string/StringStartWithFilter.java
package com.fr.bi.cal.analyze.report.report.widget.chart.newstyle.filter.string;

import com.fr.bi.cal.analyze.report.report.widget.chart.newstyle.filter.IFilter;
import com.fr.json.JSONArray;
import com.fr.json.JSONException;

/**
 * Created by AstronautOO7 on 2016/9/19.
 */
public class StringStartWithFilter implements IFilter{

    private String start;

    public StringStartWithFilter(String end) {
        this.start = end;
    }

    private boolean isStringStartWith(String value){
        if (value == null && this.start == null) {
            return true;
        }
        if (value == null || this.start == null) {
            return false;
        }
        return value.startsWith(this.start);
    }

    @Override
    public JSONArray getFilterResult(JSONArray array) throws JSONException {
        JSONArray result = new JSONArray();
        for(int i = 0; i < array.length(); i++){
            if(isStringStartWith(array.getString(i))){
                result.put(array.getString(i));
            }
        }
        return result;
    }
}

File: web/fbi_web/src/com/fr/bi/web/js/data/filter/string/string.startwith.filter.js
;
!(function(){
    BI.StringStartWithFilterValue = function(start){
        this.start = start;
    };
    BI.StringStartWithFilterValue.prototype = {
        constructor: BI.StringStartWithFilterValue,

        isStringStartWith: function(value){
            if (value == null && this.start == null) {
                return true;
            }
            if (value == null || this.start == null) {
                return false;
            }
            return value.startWith(this.start);
        },

        getFilterResult: function(array) {
            var self = this;
            return BI.filter(array, function(idx, val){
                return self.isStringStartWith(val);
            });
        }
    }
})();